apps/seqan_flexbar/general_processing.h

```diff
@@ -483,474 +483,240 @@ void processN(TSeqs& seqs, TIds& ids, TSeqs& seqsRev, TIds& idsRev, TMulti& mult
-                erase(seqs[i], 0, head);
-                if (length(seqs[i]) >= min)
-                {
-                    rem[i] = false;
-                }
-                else
-                {
-                    rem[i] = true;
-                }
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    else if (tail > 0)
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-            if (length(seqs[i]) > tail)
-            {
-                erase(seqs[i], length(seqs[i]) - tail, length(seqs[i]));
-                if (length(seqs[i]) >= min)
-                {
-                    rem[i] = false;
-                }
-                else
-                {
-                    rem[i] = true;
-                }
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    else
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-            if (length(seqs[i]) >= min)
-            {
-                rem[i] = false;
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    unsigned  ex = 0;
-    for (int j = length(rem) - 1; j >= 0; --j)
-    {
-        if (rem[j])
-        {
-            swap(seqs[j], seqs[limit - ex - 1]);
-            swap(ids[j], ids[limit - ex - 1]);
-            ++ex;
-        }
-    }
-    if (ex != 0)
-    {
-        resize(seqs, limit - ex);
-        resize(ids, limit - ex);
-        stats.removedSeqsShort += ex;
-    }
+	assert(!seqsVector.empty() && !idsVector.empty());
+	assert(seqsVector.size() == idsVector.size());
+	unsigned ex = 0;
+	unsigned limit = length(**seqsVector.begin());
+	bool demultiplex = (length(demultiplexParams.multiplexFile) > 0);
+	for (int j = limit - 1; j >= 0; j--)
+	{
+		if (rem[j])
+		{
+			for (auto &it : seqsVector) { swap((*it)[j], (*it)[limit - ex - 1]); };
+			for (auto &it : idsVector) { swap((*it)[j], (*it)[limit - ex - 1]); };
+			if (demultiplex)
+				swap(demultiplexParams.multiplex[j], demultiplexParams.multiplex[limit - ex - 1]);
+			++ex;
+		}
+	}
+	if (ex != 0)
+	{
+		for (auto &it : seqsVector) { resize(*it, limit - ex); };
+		for (auto &it : idsVector) { resize(*it, limit - ex); };
+		if (demultiplex)
+			resize(demultiplexParams.multiplex, limit - ex);
+		stats.removedSeqsShort += ex;
+	}
 }
 
-//Overload for single end data with multiplex barcodes
-template<typename TSeqs, typename TIds, typename TMulti>
-void preTrim(TSeqs& seqs, TIds& ids, TMulti& multiplex, unsigned head, unsigned tail, unsigned min, GeneralStats& stats)
+// main preTrim function
+template<typename TSeqs, typename TIds>
+void _preTrim(TSeqs& seqs, TIds& ids, unsigned head, unsigned nexus, unsigned tail, unsigned min, StringSet<bool>& rem)
```

Comment:
Please use a typename TRem. By the way, StringSet works over container types only, e.g. ```StringSet<String<bool>>```. Do you mean ```String<bool>```?